api/transport/handler.go

```diff
@@ -119,62 +115,50 @@ type StreamHandler interface {
 
 // DispatchUnaryHandler calls the handler h, recovering panics and timeout errors,
 // converting them to yarpc errors. All other errors are passed trough.
+// Deprecated: Use InvokeUnaryHandler instead.
 func DispatchUnaryHandler(
 	ctx context.Context,
 	h UnaryHandler,
 	start time.Time,
 	req *Request,
 	resq ResponseWriter,
 ) (err error) {
-	defer func() {
-		if r := recover(); r != nil {
-			log.Printf("Unary handler panicked: %v\n%s", r, debug.Stack())
-			err = fmt.Errorf("panic: %v", r)
-		}
-	}()
-
-	err = h.Handle(ctx, req, resq)
-
-	// The handler stopped work on context deadline.
-	if err == context.DeadlineExceeded && err == ctx.Err() {
-		deadline, _ := ctx.Deadline()
-		err = yarpcerrors.Newf(
-			yarpcerrors.CodeDeadlineExceeded,
-			"call to procedure %q of service %q from caller %q timed out after %v",
-			req.Procedure, req.Service, req.Caller, deadline.Sub(start))
-	}
-	return err
+	return InvokeUnaryHandler(UnaryInvokeRequest{
+		Context:        ctx,
+		StartTime:      start,
+		Request:        req,
+		ResponseWriter: resq,
+		Handler:        h,
+	},
+	)
```

Comment:
style nit: end the `)` on the same line as the `}` if it opened on the same line as the `{`